mono/metadata/sgen-stw.c

```diff
@@ -256,23 +256,49 @@ sgen_unified_suspend_stop_world (void)
 	mono_threads_begin_global_suspend ();
 	THREADS_STW_DEBUG ("[GC-STW-BEGIN][%p] *** BEGIN SUSPEND *** \n", mono_thread_info_get_tid (mono_thread_info_current ()));
 
-	FOREACH_THREAD_EXCLUDE (info, MONO_THREAD_INFO_FLAGS_NO_GC) {
-		info->client_info.skip = FALSE;
-		info->client_info.suspend_done = FALSE;
+	for (MonoThreadSuspendPhase phase = MONO_THREAD_SUSPEND_PHASE_MIN; phase < MONO_THREAD_SUSPEND_PHASE_COUNT; phase++) {
```

Comment:
I don't understand how this handle the case when we're only doing preemptive suspend.